dlmalloc_nonreuse.c

```diff
@@ -2276,6 +2275,10 @@ static void unlink_small_chunk(mstate M, mchunkptr P, size_t S) {
   else {
     CORRUPTION_ERROR_ACTION(M);
   }
+  /* Don't leak internal pointers */
+  P->pad = NULL;
+  P->fd = NULL;
+  P->bk = NULL;
```

Comment:
In the `dlfree_internal` case this is done with `memset(p, 0, CHUNK_HEADER_OFFSET);`.

Reply:
... which itself calls `unlink_chunk`, so this would end up doing the same work twice on that path.